### PR TITLE
Dataframe fixes

### LINKED ISSUE #588: Dataframe input loading strange default data (and other issues)
**Describe the bug**

The `dataframe` input loads very specific input data by default (the first column are the names "Frank", "Beatrice", "Simone", then there are two columns of numbers, as shown in the screenshot below)

image

This happens whether or not the `default` parameter is passed in. If this parameter is passed in, it is ignored, which can lead to issues, as can be seen by going to the "Working with DataFrames and Graphs" section in the getting started: https://gradio.app/getting_started/

The correct behavior should be to load the `default` dataframe if it is provided. Otherwise, instantiate with an empty dataframe.

**To Reproduce**

Minimal example:

```python
import gradio as gr

iface = gr.Interface(
    lambda x:x,
    gr.inputs.Dataframe(
        headers=["Name", "Jan Sales", "Feb Sales", "Mar Sales"],
        default=[["Jon", 12, 14, 18], ["Alice", 14, 17, 2], ["Sana", 8, 9.5, 12]],
    ),
    "dataframe",
)
iface.launch()
```

### PR DESCRIPTION
**Pull request by pngwn**
- default data is now respected
- Backspace and Delete can clear cells
- Heading label has been removed

Closes #588

### CODE REVIEW

File: ui/packages/app/src/components/input/DataFrame/DataFrame.svelte [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-	];
+	export let values: Array<Array<string | number>> = [["", "", ""]];
+	export let default_data: Array<Array<string | number>> = [];
+
+	if ($$props.default) values = $$props.default;
 
 	export let setValue: (val: typeof values) => typeof values;
 	export let editable = true;
 
+	$: console.log($$props);
```
Comment (aliabid94) [inline ui/packages/app/src/components/input/DataFrame/DataFrame.svelte:13]: remove?

File: ui/packages/app/src/components/input/DataFrame/DataFrame.svelte [hunk 2]
```diff
@@ -87,6 +87,7 @@
 		id: number
 	) {
 		let is_data;
+		console.log(event.key, data);
```
Comment (aliabid94) [inline ui/packages/app/src/components/input/DataFrame/DataFrame.svelte:90]: remove?

**Review by aliabid94 — APPROVED** [review]
LGTM

**Comment by aliabid94** [comment]
Do you know why this only ran the static checks, not the entire circleci build @pngwn @FarukOzderim ?

**Comment by FarukOzderim** [comment]
Maybe since only ui folder has changed :D? I checked other commits only affecting the ui folder but they have circleci tests 😕 
Probably a bug on circleci side, maybe it was down at the time of the commit.

**Comment by FarukOzderim** [comment]
While checking $$ syntax I came across this, is this an important point @pngwn?

`$$props references all props that are passed to a component, including ones that are not declared with export. It is not generally recommended, as it is difficult for Svelte to optimise. But it can be useful in rare cases – for example, when you don't know at compile time what props might be passed to a component.`

**Comment by pngwn** [comment]
@FarukOzderim Yes it is. In general the more specific we are with references, the more svelte can optimise the code. But things like `$$props` are useful when debugging to take a look at all props being passed around.

They have been removed now.

**Comment by pngwn** [comment]
Going to merge this and create a new issues for removing rows/ cols.

**Pull request merged**